src/plugins/flare/flareplugin.cpp

```diff
@@ -367,10 +394,13 @@ bool FlarePlugin::write(const Tiled::Map *map, const QString &fileName)
                     out << "," << w << "," << h << "\n";
 
                     // write all properties for this object
-                    Properties::const_iterator it = o->properties().constBegin();
-                    Properties::const_iterator it_end = o->properties().constEnd();
-                    for (; it != it_end; ++it) {
-                        out << it.key() << "=" << it.value().toString() << "\n";
+                    QVariantMap propsMap = o->properties();
+                    for (QVariantMap::const_iterator it = propsMap.constBegin(); it != propsMap.constEnd(); ++it) {
+                        if (it->userType() == filePathTypeId()) {
+                            out << it.key() << "=" << mapDir.relativeFilePath(toExportValue(it.value()).toString()) << "\n";
+                        } else {
+                            out << it.key() << "=" << toExportValue(it.value()).toString() << "\n";
+                        }
```

Comment:
On closer inspection, I believe this could be simplified to just:
```c++
out << it.key() << "=" << toExportValue(it.value(), mapDir).toString() << "\n";
```
The same goes for the writing of layer properties.

The version of `toExportValue` taking an additional map directory is new and will handle the file path case.

Reply:
You're right. That syntax is even used on line 324. I fixed the two spots that needed it.